tests/cases/compiler/satisfiesScenarios.ts

```diff
@@ -0,0 +1,112 @@
+
+const x = [1, 2] satisfies [number, number];
+
+interface A {
+    a: string
+}
+let d = { a: 'test' } satisfies A;
+let e = { a: 'test', b: 'test' } satisfies A;
+
+// Most examples from #47920
+namespace PropNameConstraining {
+    type Keys = 'a' | 'b' | 'c' | 'd';
+
+    const p = {
+        a: 0,
+        b: "hello",
+        x: 8 // Should error, 'x' isn't in 'Keys'
+    } satisfies Partial<Record<Keys, unknown>>;
+
+    // Should be OK -- retain info that a is number and b is string
+    let a = p.a.toFixed();
+    let b = p.b.substring(1);
+    // Should error even though 'd' is in 'Keys'
+    let d = p.d;
+}
+
+namespace PropertyNameFulfillment {
+    type Keys = 'a' | 'b' | 'c' | 'd';
+
+    const p = {
+        a: 0,
+        b: "hello",
+        x: 8 // Should error, 'x' isn't in 'Keys'
+    } satisfies Record<Keys, unknown>;
+
+    // Should be OK -- retain info that a is number and b is string
+    let a = p.a.toFixed();
+    let b = p.b.substr(1);
+    // Should error even though 'd' is in 'Keys'
+    let d = p.d;
+}
+
+namespace PropertyValueConformance {
+    type Facts = { [key: string]: boolean };
+    declare function checkTruths(x: Facts): void;
+    declare function checkM(x: { m: boolean }): void;
+    const x = {
+        m: true
+    };
+
+    // Should be OK
+    checkTruths(x);
+    // Should be OK
+    checkM(x);
+    // Should fail under --noIndexSignaturePropertyAccess
+    console.log(x.z);
+    // Should be OK under --noUncheckedIndexedAccess
+    const m: boolean = x.m;

```

Comment:
Do you have a toggled version of this test at all to test these scenarios?

Reply:
I think call doesn't directly affect `satisfies` usage. Anyway I've added two tests `typeSatisfaction_propertyValueConformance1`, `typeSatisfaction_propertyValueConformance1` with `noPropertyAccessFromIndexSignature: true, false` and `@noUncheckedIndexedAccess: true, false`